From aa969fd8307c30a96ac42439d57a642bbdb613df Mon Sep 17 00:00:00 2001
From: Calum Lind <calumlind+deluge@gmail.com>
Date: Sat, 6 Oct 2012 22:54:33 +0100
Subject: [PATCH] Fix #2169 : Add Torrent Dialog 'Download Location' not set
 correctly when folder typed into Other->Location field

Using get_current_folder() in the file chooser will use the default folder when a folder path has been typed into the Location field rather than selected with mouse. The solution is to replace it with get_filename().
---
 deluge/plugins/extractor/extractor/gtkui.py | 4 ++--
 deluge/ui/gtkui/addtorrentdialog.py         | 2 +-
 deluge/ui/gtkui/options_tab.py              | 2 +-
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/deluge/plugins/extractor/extractor/gtkui.py b/deluge/plugins/extractor/extractor/gtkui.py
index 9e3bf61..bce7667 100644
--- a/deluge/plugins/extractor/extractor/gtkui.py
+++ b/deluge/plugins/extractor/extractor/gtkui.py
@@ -65,7 +65,7 @@ def disable(self):
     def on_apply_prefs(self):
         log.debug("applying prefs for Extractor")
         if client.is_localhost():
-            path = self.glade.get_widget("folderchooser_path").get_current_folder()
+            path = self.glade.get_widget("folderchooser_path").get_filename()
         else:
             path = self.glade.get_widget("entry_path").get_text()
 
@@ -89,7 +89,7 @@ def on_get_config(config):
                 self.glade.get_widget("folderchooser_path").set_current_folder(config["extract_path"])
             else:
                 self.glade.get_widget("entry_path").set_text(config["extract_path"])
-            
+
             self.glade.get_widget("chk_use_name").set_active(config["use_name_folder"])
 
         client.extractor.get_config().addCallback(on_get_config)
diff --git a/deluge/ui/gtkui/addtorrentdialog.py b/deluge/ui/gtkui/addtorrentdialog.py
index 32bc0be..a26acf0 100644
--- a/deluge/ui/gtkui/addtorrentdialog.py
+++ b/deluge/ui/gtkui/addtorrentdialog.py
@@ -405,7 +405,7 @@ def save_torrent_options(self, row=None):
 
         if client.is_localhost():
             options["download_location"] = \
-                self.glade.get_widget("button_location").get_current_folder()
+                self.glade.get_widget("button_location").get_filename()
         else:
             options["download_location"] = \
                 self.glade.get_widget("entry_download_path").get_text()
diff --git a/deluge/ui/gtkui/options_tab.py b/deluge/ui/gtkui/options_tab.py
index 6ea7691..8e9eaef 100644
--- a/deluge/ui/gtkui/options_tab.py
+++ b/deluge/ui/gtkui/options_tab.py
@@ -179,7 +179,7 @@ def _on_button_apply_clicked(self, button):
             client.core.set_torrent_move_completed(self.prev_torrent_id, self.chk_move_completed.get_active())
         if self.chk_move_completed.get_active():
             if client.is_localhost():
-                path = self.filechooser_move_completed.get_current_folder()
+                path = self.filechooser_move_completed.get_filename()
             else:
                 path = self.entry_move_completed.get_text()
             client.core.set_torrent_move_completed_path(self.prev_torrent_id, path)
-- 
1.8.0

